extensions/binding/org.eclipse.smarthome.binding.dmx.test/src/test/java/org/eclipse/smarthome/binding/dmx/test/TestBridgeHandler.java

```diff
@@ -88,12 +88,7 @@ public void initialize() {
     public long calcBuffer(long time, long timespan) {
         universe.calculateBuffer(time);
         universe.calculateBuffer(time + timespan);
-        try {
-            // needed for the framework to process all events
-            Thread.sleep(200);
-        } catch (InterruptedException e) {
-            logger.error("failed to sleep for 200 ms");
-        }
+        logger.debug("calculated buffer for {} to {} (difference {})", time, time + timespan, timespan);
```

Comment:
Now you have just removed the waiting time but before you argued that the framework needs time to process your events.

So you probably have to wait until the event is fully processed at the place where you call `calcBuffer`. Am I right?

You can find an example for the `waifForAssert` [here](https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/items/GroupItemTest.java#L104)

Reply:
This was a leftover from an earlier attempt to fix the randomly failing tests. After some more investigation, this real problem is that sometimes the updates from the things seem to disappear. SInstead a waitforAssert is implemented in all tests, waiting for the thing channel update.